src/Microsoft.AspNet.OData.Shared/Builder/ODataConventionModelBuilder.cs

```diff
@@ -694,41 +698,52 @@ private void MapStructuralProperty(StructuralTypeConfiguration type, PropertyInf
 
         // figures out the type of the property (primitive, complex, navigation) and the corresponding edm type if we have seen this type
         // earlier or the user told us about it.
-        private PropertyKind GetPropertyType(PropertyInfo property, out bool isCollection, out IEdmTypeConfiguration mappedType)
+        private PropertyKind GetPropertyType(PropertyInfo property, out bool isCollection, out IEdmTypeConfiguration mappedType, out bool IsValid)
         {
             Contract.Assert(property != null);
 
+            IsValid = true;
+            mappedType = null;
+            isCollection = false;
+
+            if (typeof(ODataIdContainer) == property.PropertyType)
+            {
+                IsValid = false;
+                return (PropertyKind)(int.MaxValue);
+            }
+
             // IDictionary<string, object> is used as a container to save/retrieve dynamic properties for an open type.
             // It is different from other collections (for example, IEnumerable<T> or IDictionary<string, int>)
             // which are used as navigation properties.
             if (typeof(IDictionary<string, object>).IsAssignableFrom(property.PropertyType))
             {
-                mappedType = null;
-                isCollection = false;
                 return PropertyKind.Dynamic;
             }
 
             // IODataInstanceAnnotationContainer is used as a container to save/retrieve instance annotation properties for a CLR type.
             // It is different from other collections (for example, IDictionary<string,IDictionary<string, int>>)          
             if (typeof(IODataInstanceAnnotationContainer).IsAssignableFrom(property.PropertyType))
             {
-                mappedType = null;
-                isCollection = false;
-
                 return PropertyKind.InstanceAnnotations;
             }
 
             PropertyKind propertyKind;
             if (TryGetPropertyTypeKind(property.PropertyType, out mappedType, out propertyKind))
             {
-                isCollection = false;
                 return propertyKind;
             }
 
             Type elementType;
             if (TypeHelper.IsCollection(property.PropertyType, out elementType))
             {
                 isCollection = true;
+
```

Comment:
yikes -- there should never be a reason for having a collection of ODataIdContainers, but doesn't hurt to check for it.